python/eos/plot/plotter.py

```diff
@@ -99,15 +99,19 @@ def setup_plot(self):
         if not 'plot' in self.instructions:
             raise KeyError('no plot metadata specified')
 
-        myplot = self.instructions['plot']
+        if 'axis' in self.instructions:
+            self.ax = self.instructions['axis']
```

Comment:
This leaves ``self.fig`` uninitialized. Can you at least set it to ``None``?

Reply:
This should be done here:

https://github.com/eos/eos/pull/617/files#diff-5ac706e3ba490789c67c7e29957bb047c2ba97e7f97d8e316636331751102b01R42

Reply:
You are absolutely right! I wave my concerns. I think it can be merged now, if all of @mreboud's concerns are removed, too!